File: Java/sevenwonders.java

//  https://open.kattis.com/problems/sevenwonders
import java.util.Arrays;
import java.util.Scanner;

public class sevenwonders {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String letters = sc.nextLine();
        int[] counts = { 0, 0, 0 };
        for (int x = 0; x < letters.length(); x++) {
            if (letters.charAt(x) == 'T')
                counts[0]++;
            else if (letters.charAt(x) == 'C')
                counts[1]++;
            else
                counts[2]++;
        }
        Arrays.sort(counts);
        System.out.println(
                (int) (Math.pow(counts[0], 2) + Math.pow(counts[1], 2) + Math.pow(counts[2], 2) + (counts[0] * 7)));
        sc.close();
    }
}

File: C Sharp/sevenwonders.cs
using System;

public class sevenwonders {
    public static void Main (string[] args) {
        int t = 0;
        int c = 0;
        int g = 0;
        string line = Console.ReadLine ();
        for (int x = 0; x < line.Length; x++) {
            if (line[x] == 'T')
                t++;
            else if (line[x] == 'C')
                c++;
            else
                g++;
        }
        int min = t <= c && t <= g ? t : c <= t && c <= g ? c : g;
        Console.WriteLine ((int) (Math.Pow (t, 2) + Math.Pow (c, 2) + Math.Pow (g, 2) + (min * 7)));
    }
}